From 3880b1507ca15192078fbf9fcc4bc2b43892918b Mon Sep 17 00:00:00 2001
From: Jan Pobrislo <ccx@webprojekty.cz>
Date: Tue, 11 Aug 2009 14:10:21 +0200
Subject: [PATCH] BUG(2256): Make running ldconfig optional

---
 wscript |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/wscript b/wscript
index d23e0bc..cd39ea8 100644
--- a/wscript
+++ b/wscript
@@ -486,13 +486,20 @@ def set_options(opt):
                    help="Force a specific Windows version (cross-compilation)")
     opt.add_option('--run-tests', action='store_true', default=False,
                    dest='run_tests', help="Run test suite")
+    opt.add_option('--with-ldconfig', action='store_true', default=None,
+                   dest='ldconfig', help="Run ldconfig after install even if not root")
+    opt.add_option('--without-ldconfig', action='store_false',
+                   dest='ldconfig', help="Don't run ldconfig after install")
 
     opt.sub_options("src/xmms")
     for o in optional_subdirs + subdirs:
         opt.sub_options(o)
 
 def shutdown():
-    if Options.commands['install'] and os.geteuid() == 0:
+    if Options.commands['install'] and (
+            Options.options.ldconfig or
+            (Options.options.ldconfig is None and os.geteuid() == 0)
+            ):
         ldconfig = '/sbin/ldconfig'
         if os.path.isfile(ldconfig):
             libprefix = Utils.subst_vars('${PREFIX}/lib', Build.bld.env)
-- 
1.6.5.rc1

