--- CMakeLists.txt.orig	2008-01-04 00:58:48.000000000 +0400
+++ CMakeLists.txt	2008-10-04 01:08:33.810374635 +0500
@@ -332,7 +332,7 @@ IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin"
 
 ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
-  SET(INSTALL_SUBDIR_BIN "games/")
+  SET(INSTALL_SUBDIR_BIN "games/bin/")
   SET(INSTALL_SUBDIR_SHARE "share/games/supertux2/")
   SET(INSTALL_SUBDIR_DOC "share/doc/supertux2/")
 
