Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 5784)
+++ CMakeLists.txt	(working copy)
@@ -130,8 +130,10 @@
 ## Find revision of WC
 FIND_PACKAGE(Subversion)
 IF(Subversion_FOUND)
-  Subversion_WC_INFO(${SUPERTUX_SOURCE_DIR} SUPERTUX)
-  SET(REVISION " r${SUPERTUX_WC_REVISION}")
+  IF(EXISTS ${SUPERTUX_SOURCE_DIR}/.svn)
+    Subversion_WC_INFO(${SUPERTUX_SOURCE_DIR} SUPERTUX)
+    SET(REVISION " r${SUPERTUX_WC_REVISION}")
+  ENDIF(EXISTS ${SUPERTUX_SOURCE_DIR}/.svn)
 ENDIF(Subversion_FOUND)
 
 ## Check platform-dependent build options
