Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 6252)
+++ CMakeLists.txt	(working copy)
@@ -147,11 +147,6 @@
 
 INCLUDE(ConfigureChecks)
 
-## Create config.h and version.h
-
-configure_file(config.h.cmake ${CMAKE_BINARY_DIR}/config.h )
-include_directories (${CMAKE_BINARY_DIR}/)
-
 ## Also execute instructions in src/squirrel/CMakeLists.txt
 
 ADD_SUBDIRECTORY(external/squirrel)
@@ -443,6 +438,11 @@
 ENDIF(WIN32 AND NOT UNIX)
 INCLUDE(CPack)
 
+## Create config.h and version.h
+
+configure_file(config.h.cmake ${CMAKE_BINARY_DIR}/config.h )
+include_directories (${CMAKE_BINARY_DIR}/)
+
 # move some config clutter to the advanced section
 MARK_AS_ADVANCED(
   INSTALL_SUBDIR_BIN
