Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 5784)
+++ CMakeLists.txt	(working copy)
@@ -384,7 +386,7 @@
 
 ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
-  SET(INSTALL_SUBDIR_BIN "games/")
+  SET(INSTALL_SUBDIR_BIN "games/bin/")
   SET(INSTALL_SUBDIR_SHARE "share/games/supertux2/")
   SET(INSTALL_SUBDIR_DOC "share/doc/supertux2/")
 
